### DISCUSSION TITLE
Load local drawable or svg using Coil3

### DISCUSSION THREAD

**Discussion by DawnNguyenAhiho** [coil-kt/coil#2767] [Q&A]
Hi everyone, I'm trying to load local svg drawable using coil in my Jetpack Compose android native porject.
For context, my app has some default image and allow user to upload their own image and I will save the uploaded image to dataDir. I have to save the uri to the image to SQLite database so that I can show it to user. So that, instead of using DrawRes (`R.drawable.`), I use the URI (`android.resource://my.package.name/drawable/mydrawable`) to load and show image, then use that uri to save to DB. 
But after this [PR](https://github.com/coil-kt/coil/pull/2294), I'm unable to load SVG drawable using URI anymore. After reading docs, I saw that I can use coil-svg to load svg using coil. The problem is whatever path I pass to AsyncImage, it won't load the svg for me, only the `R.drawable` or `R.raw.mysvg` work. Are there anyway to load SVG using path, so that I won't have to change much at my current logic

**Comment by DawnNguyenAhiho** [comment]
I'm currently write this Mapper, not sure does it prevent resource shrinking optimization (which lead to the PR above). Hope some one can correct me
```
internal class DrawResUriMapper: Mapper<String, Int> {
    override fun map(data: String, options: Options): Int? {
        val data = data.toUri()
        if (!isApplicable(data)) return null

        return when (val drawableName = data.pathSegments[1]) {
            "drawable_name" -> R.drawable.drawable_name
            else -> {
                if (BuildConfig.BUILD_TYPE == "debug") {
                    throw IllegalArgumentException("Unknown drawable name: $drawableName")
                } else {
                    return null
                }
            }
        }
    }

   private fun isApplicable(data: Uri): Boolean {
       return data.scheme == SCHEME_ANDROID_RESOURCE &&
               !data.authority.isNullOrBlank() &&
               data.pathSegments.count() == 2
               && data.pathSegments[0] == "drawable"
   }
}
```

**Reply by Mortalite** [reply]
You can simply add this mapper back in:
```
class CustomApplication : SingletonImageLoader.Factory {
    override fun newImageLoader(context: Context): ImageLoader =
       ImageLoader.Builder(context)
            .components {
                add(ResourceUriMapper())
            }
            .build()
}
```

**Reply by DawnNguyenAhiho** [reply]
Yeah, but that PR and also the Android Studio lint show that the solution in `ResourceUriMapper` will prevent R8 to optimize and shrink resource, I would rather not use that and seeking for other solution

**Reply by colinrtwhite** [reply]
Any usage of `resources.getIdentifier(name, type, packageName)` will break the optimization. Don't use it and it won't break the optimization.

**Reply by DawnNguyenAhiho** [reply]
Yes, I understand that so I won't use any `resource.getIdentifier`. Instead, I use custom URI and write Custom Mapper and map the URI to drawable by ` "drawable_name" -> R.drawable.drawable_name`. I want to ask is it ok to use
